src/main/resources/assets/sodium/shaders/blocks/block_layer_opaque.vsh

```diff
@@ -22,7 +23,7 @@ void main() {
     vec3 position = u_RegionOffset + _draw_translation + _vert_position;
 
 #ifdef USE_FOG
-    v_FragDistance = max(length(position.xz), abs(position.y)); // Use the maximum of the horizontal and vertical distance to get cylindrical fog
+    v_FragDistance = u_FogShape == 0 ? length(position) : max(length(position.xz), abs(position.y)); // Use the maximum of the horizontal and vertical distance to get cylindrical fog if fog shape is 1 (cylindrical)
```

Comment:
Can `u_FogShape` be promoted to a compile time define? The conditional is unlikely to matter a ton but it seems like a minimal change to avoid it all together.

Reply:
I don’t believe so, as it can be changed by gameplay conditions such as going into water.